meta_review/handler.py

```diff
@@ -0,0 +1,499 @@
+                continue
+            for comment in issue['pullRequest']['comments']:
+                # parse time
+                comment['createdAt'] = parse_time(comment['createdAt'])
+                comment['lastEditedAt'] = parse_time(comment['lastEditedAt'])
+                self.comments[comment['id']] = comment
+
+        # save reactions in memory
+        self.reactions = dict()
+        for key, comment in self.comments.items():
+            if not comment.get('reactions'):
+                continue
+            for reaction in comment['reactions']:
+                # record receiver
+                reaction['receiver'] = {
+                    'login': comment['author']['login']
+                }
+                # record comment id
+                reaction['comment_id'] = comment['id']
+                # parse time
+                reaction['createdAt'] = parse_time(reaction['createdAt'])
+                self.reactions[reaction['id']] = reaction
+
+        # save participants in memory
+        self.participants = dict()
+        for key, comment in self.comments.items():
+            # get author of each comment
+            author = comment['author']
+
+            # skip if user does not exist
+            # this happens when account is deleted from GitHub
+            if author['login']:
+                self.participants[author['login']] = author
+
+            if not comment.get('reactions'):
+                continue
+            for reaction in comment['reactions']:
+                # get user of each reaction
+                user = reaction['user']
+
+                # skip if user does not exist
+                if user['login']:
+                    self.participants[user['login']] = user
+
+    def handle(self):
+        '""Scrape data, process and store in database.""'
+        is_first_deploy = self._is_first_deploy()
+
+        load_all(self.participants, self.comments, self.reactions)
+        dump_all(self.participants, self.comments, self.reactions)
+
+        self._update_score()
+        self._update_weight_factors()
+
+        if is_first_deploy:
+            # dump first to make use of reverse query
+            dump_all(self.participants, self.comments, self.reactions)
+            self._recalculate_score()
+
+        self._punish_improper_behavior()
+        self._update_time()
+
+        # dump first to make use of built-in sort method
+        dump_all(self.participants, self.comments, self.reactions)
+        self._update_rankings()
+
+        dump_all(self.participants, self.comments, self.reactions)
+
+        self.logger.info('Meta Review System finishes.')
+
+    def _punish_improper_behavior(self):
+        """
+        Give punishment points to participants who did improper behavior.
+
+        If an author updates his review comment after it has been
+        meta-reviewed, he will be slightly punished.
+        """
+        for key, comment in self.comments.items():
```

Comment:
This is looking better, but it still mixes analysis algorithms with business logic.

Create an extra public method which returns a generator of the modified-after-meta-review comments.  Then it is able to be documented as a dataset, and the documentation will also help it be reused because other people know about this list of people.

You could even have another method which converts the list of 'modified-after-meta-review comments' into a dict of `{author: count_of_edited_comments_after_meta_review}` (or adds `author.count_of_edited_comments_after_meta_review`?), and then here you can very easily log and update those authors.